doc/user/content/overview/key-concepts.md

```diff
@@ -190,20 +191,63 @@ encoded in Avro with the [Debezium diff envelope](/sql/create-sink#debezium-enve
 
 ## Clusters
 
-{{< experimental >}}
-The concept of a cluster
-{{< /experimental >}}
+Clusters are logical components that let you express resource isolation for all
+dataflow-powered objects, e.g. indexes and sinks. When creating dataflow-powered
+objects, you must specify which cluster you want to use. (Not naming a cluster
+explicitly uses your session's default cluster.)
+
+Importantly, clusters are strictly a logical component; they rely on  [cluster
+replicas](#cluster-replicas) to run dataflows. Said a slightly different way, a
+cluster with no replicas does no computation. For example, if you create an
+index on a cluster with no replicas, you cannot select from that index because
+there is no physical representation of the index to read from.
+
+Though clusters only represent the logic of which objects you want to bundle
+together, this impacts the performance characteristics once you provision
+cluster replicas. Each object in a cluster gets instantiated on every replica,
+meaning they are all in contention for the same physical resources. To achieve
```

Comment:
```suggestion
meaning that on a given physical replica, are objects in the cluster are in contention for the same physical resources. To achieve
```

The text as written is a little ambiguous IMO. My suggestion is choppy but demonstrates the intent!